MekHQ/src/mekhq/campaign/parts/equipment/AmmoBin.java

```diff
@@ -490,8 +490,9 @@ public boolean isSamePartType(Part part) {
         // AmmoType and number of rounds of ammo (i.e. they are the same
         // irrespective of "munition type" or "bomb type").
         return (getClass() == part.getClass()) &&
-               getType().isCompatibleWith(((AmmoBin) part).getType()) &&
-               (((AmmoBin) part).getFullShots() == getFullShots());
+            ((getType().isCompatibleWith(((AmmoBin) part).getType())) ||
+                   (getType().equals(((AmmoBin) part).getType()))) &&
```

Comment:
OMFG :face_holding_back_tears:

Reply:
Sorry, just wanted to echo Illiani's comment.  That must have been super-annoying to track down.